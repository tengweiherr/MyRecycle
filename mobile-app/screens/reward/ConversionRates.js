import React, { useContext, useEffect, useState } from "react";
import { NativeBaseProvider, Box, Button, ScrollView, Stack, Center, Accordion, Text, Divider, List, Image, PresenceTransition, Pressable, VStack, Spinner, HStack, InfoOutlineIcon } from 'native-base';
import { DataTable } from 'react-native-paper';
import { useNavigation } from '@react-navigation/native';
import { ImageBackground } from "react-native";
import { API_URL } from "@env";
import { LoadingContext } from "../../context/context";


export const ConversionRates = () => {
  
  const Navigation = useNavigation(); 

  const { setIsLoading } = useContext(LoadingContext);

  const [data, setData] = useState(null);
  const [error, setError] = useState("");

  //load reward data
  useEffect(() => {

    setIsLoading(true);

    fetch(API_URL + 'material/', {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then(response => response.json())
      .then(results => {

        let temp = [];
        let count = 0;
        //get only recyclables
        for (let index = 0; index < results.length; index++) {
          if(results[index].category !== "Non-Recyclable"){
            temp[count] = results[index];
            count++;
          }
        }

        setData(temp);
        setIsLoading(false);
      })
      .catch(err => {
        setIsLoading(false);
        setError(err);
      });

  }, []);

  const _renderMaterial = () => {
    return data.map((item, index) => {
      return (
        <DataTable.Row key={index} style={index%2!==0 && {backgroundColor:"#f8f8f8"}}>
          <DataTable.Cell>{index === 0 && <Text>Per kilogram</Text>}</DataTable.Cell>
          <DataTable.Cell>{item.material}</DataTable.Cell>
          <DataTable.Cell numeric>{item.conversion_rate}</DataTable.Cell>
        </DataTable.Row>
      )
    })
  }

  return (
    <ImageBackground source={require("../../public/assets/bg-1-100.jpg")} style={{width:"100%", height:"100%"}}>
      <ScrollView>
      <VStack p="5" space={3}>
        <>
        <DataTable backgroundColor="#fff" borderColor="#f8f8f8" style={{borderWidth:2, borderRadius:12}}>
          <DataTable.Header style={{backgroundColor:"#1FAA8F", borderTopEndRadius:12, borderTopStartRadius:12}}>
            <DataTable.Title>
              <Text color="#fff" bold fontSize="md">Unit</Text>
            </DataTable.Title>
            <DataTable.Title>
              <Text color="#fff" bold fontSize="md">Materials</Text>
            </DataTable.Title>
            <DataTable.Title numeric>
              <Text color="#fff" bold fontSize="md">MR Points</Text>
            </DataTable.Title>
          </DataTable.Header>
          {data && _renderMaterial()}
        </DataTable>
        <HStack py="2" space={3} justifyContent="flex-start" alignItems="flex-start">
          <InfoOutlineIcon size="xs" color="yellow.500"/>
            <VStack>
              <Text fontSize="xs" width="xs">
              Please be informed that the conversion rates will be reviewed and adjusted without prior notification if necessary.
              </Text>
              {/* <Pressable onPress={()=>Navigation.navigate('GeneralWaste')}>
                 <Text fontSize="xs" color="#1FAA8F" italic={true}>See here</Text>
              </Pressable> */}
            </VStack>
        </HStack>
        </>
        <Button rounded="full" background="#FC8B10" borderColor="#FC8B10" borderWidth="1" onPress={()=>Navigation.goBack()}>
          <Text testID="goback" bold color="white">Go Back</Text>
        </Button>
      </VStack>
      </ScrollView>
    </ImageBackground>
  );
};


export default () => {
    return (
      <NativeBaseProvider>
        <Center flex={1}>
          <ConversionRates />
        </Center>
      </NativeBaseProvider>
    )
};
