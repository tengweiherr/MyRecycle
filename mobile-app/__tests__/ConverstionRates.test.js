import React from 'react';
import renderer from 'react-test-renderer';
import AsyncStorage from '@react-native-async-storage/async-storage';
import ConversionRates from '../screens/reward/ConversionRates';
import { NativeBaseProvider } from "native-base";
import { enableFetchMocks } from 'jest-fetch-mock'
enableFetchMocks();

// beforeEach(async () => {
//     await AsyncStorage.clear();
//   });

  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  };

  const tree = renderer.create(
    <NativeBaseProvider initialWindowMetrics={inset}>
        <ConversionRates />
    </NativeBaseProvider>);

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));


  describe('Test', () => {

    test('testing',  () => {
        expect(tree).toMatchSnapshot();
    })

    test("go back", ()=>{
      const goback = tree.root.findByProps({testID:"goback"}).props;
      expect(goback.children).toEqual('Go Back');
    })

});