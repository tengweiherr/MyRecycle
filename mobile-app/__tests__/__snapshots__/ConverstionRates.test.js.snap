// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test testing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          dataSet={Object {}}
          style={Object {}}
        >
          <View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                }
              }
            >
              <View
                backgroundColor="#fff"
                borderColor="#f8f8f8"
                style={
                  Array [
                    Object {
                      "width": "100%",
                    },
                    Object {
                      "borderRadius": 12,
                      "borderWidth": 2,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "height": 48,
                        "paddingHorizontal": 16,
                      },
                      Object {
                        "borderBottomColor": "rgba(0, 0, 0, 0.12)",
                      },
                      Object {
                        "backgroundColor": "#1FAA8F",
                        "borderTopEndRadius": 12,
                        "borderTopStartRadius": 12,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignContent": "center",
                          "flex": 1,
                          "flexDirection": "row",
                          "paddingVertical": 12,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "fontSize": 12,
                              "fontWeight": "500",
                              "lineHeight": 24,
                            },
                            Object {
                              "maxHeight": 24,
                            },
                            Object {},
                            Object {
                              "color": "rgba(0, 0, 0, 0.6)",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#fff",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Unit
                      </Text>
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignContent": "center",
                          "flex": 1,
                          "flexDirection": "row",
                          "paddingVertical": 12,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "fontSize": 12,
                              "fontWeight": "500",
                              "lineHeight": 24,
                            },
                            Object {
                              "maxHeight": 24,
                            },
                            Object {},
                            Object {
                              "color": "rgba(0, 0, 0, 0.6)",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#fff",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Materials
                      </Text>
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignContent": "center",
                          "flex": 1,
                          "flexDirection": "row",
                          "paddingVertical": 12,
                        },
                        Object {
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "fontSize": 12,
                              "fontWeight": "500",
                              "lineHeight": 24,
                            },
                            Object {
                              "maxHeight": 24,
                            },
                            Object {},
                            Object {
                              "color": "rgba(0, 0, 0, 0.6)",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#fff",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        MR Points
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 12,
                  }
                }
              />
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 8,
                    "paddingTop": 8,
                  }
                }
              >
                <RNSVGSvgView
                  accessibilityRole="image"
                  align="xMidYMid"
                  bbHeight={12}
                  bbWidth={12}
                  color={4293571336}
                  dataSet={Object {}}
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  stroke=""
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "color": "#eab308",
                        "height": 12,
                        "width": 12,
                      },
                      Object {
                        "flex": 0,
                        "height": 12,
                        "width": 12,
                      },
                    ]
                  }
                  tintColor={4293571336}
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    propList={
                      Array [
                        "stroke",
                      ]
                    }
                    stroke={null}
                  >
                    <RNSVGGroup>
                      <RNSVGGroup
                        fill={
                          Array [
                            2,
                          ]
                        }
                        propList={
                          Array [
                            "fill",
                            "stroke",
                          ]
                        }
                        stroke={null}
                      >
                        <RNSVGPath
                          d="M11.9442 8.24C12.5628 8.24 13.0642 7.73856 13.0642 7.12C13.0642 6.50144 12.5628 6 11.9442 6C11.3257 6 10.8242 6.50144 10.8242 7.12C10.8242 7.73856 11.3257 8.24 11.9442 8.24Z"
                        />
                        <RNSVGPath
                          d="M14.4008 15.9996H12.8008V9.59961H10.4008C10.1886 9.59961 9.98513 9.68389 9.8351 9.83392C9.68507 9.98395 9.60078 10.1874 9.60078 10.3996C9.60078 10.6118 9.68507 10.8153 9.8351 10.9653C9.98513 11.1153 10.1886 11.1996 10.4008 11.1996H11.2008V15.9996H9.60078C9.38861 15.9996 9.18513 16.0839 9.0351 16.2339C8.88507 16.384 8.80078 16.5874 8.80078 16.7996C8.80078 17.0118 8.88507 17.2153 9.0351 17.3653C9.18513 17.5153 9.38861 17.5996 9.60078 17.5996H14.4008C14.613 17.5996 14.8164 17.5153 14.9665 17.3653C15.1165 17.2153 15.2008 17.0118 15.2008 16.7996C15.2008 16.5874 15.1165 16.384 14.9665 16.2339C14.8164 16.0839 14.613 15.9996 14.4008 15.9996Z"
                        />
                        <RNSVGPath
                          d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17316C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8078C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34783 20.9464 6.80429 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM12 20.3333C10.3518 20.3333 8.74066 19.8446 7.37025 18.9289C5.99984 18.0132 4.93174 16.7117 4.30101 15.189C3.67028 13.6663 3.50525 11.9908 3.82679 10.3742C4.14834 8.75774 4.94201 7.27288 6.10745 6.10744C7.27288 4.942 8.75774 4.14833 10.3743 3.82679C11.9908 3.50525 13.6663 3.67027 15.189 4.301C16.7117 4.93173 18.0132 5.99984 18.9289 7.37025C19.8446 8.74066 20.3333 10.3518 20.3333 12C20.3333 13.0943 20.1178 14.178 19.699 15.189C19.2802 16.2001 18.6664 17.1187 17.8926 17.8926C17.1187 18.6664 16.2001 19.2802 15.189 19.699C14.178 20.1178 13.0944 20.3333 12 20.3333Z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "width": 12,
                    }
                  }
                />
                <View
                  dataSet={Object {}}
                  style={Object {}}
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 18,
                        "textDecorationLine": undefined,
                        "width": 320,
                      }
                    }
                  >
                    Please be informed that the conversion rates will be reviewed and adjusted without prior notification if necessary.
                  </Text>
                </View>
              </View>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 12,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={Object {}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onHoverIn={[Function]}
                onHoverOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FC8B10",
                    "borderColor": "#FC8B10",
                    "borderRadius": 9999,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={Object {}}
                    style={Object {}}
                  >
                    <Text
                      dataSet={Object {}}
                      style={
                        Object {
                          "backgroundColor": undefined,
                          "color": "#FFFFFF",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "700",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                      testID="goback"
                    >
                      Go Back
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RNCSafeAreaProvider>
</RNCSafeAreaProvider>
`;
